executor-service/src/main/java/executor/service/maintenance/plugin/proxy/ProxySourcesClientImpl.java

```
@@ -0,0 +1,55 @@
+package executor.service.maintenance.plugin.proxy;
+
+import executor.service.model.ProxyConfigHolderDto;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class ProxySourcesClientImpl implements ProxySourcesClient {
+
+    private final ProxyValidator mValidator;
+    private final List<ProxyConfigHolderDto> proxyConfigHolders;
+    private Iterator<ProxyConfigHolderDto> mIterator;
+    boolean infinityLooping = false;
+
+    public void setInfinityLooping(boolean value) {
+        //todo Is it need this option?
+        infinityLooping = value;
+    }
+
+    public ProxySourcesClientImpl(ProxySources sources) {
+        this(sources, null);
+    }
+
+    public ProxySourcesClientImpl(ProxySources sources, ProxyValidator validator) {
+        proxyConfigHolders = sources.getProxyConfigHolders();
+        mValidator = validator;
+        mIterator = proxyConfigHolders.iterator();
+    }
+
+    private ProxyConfigHolderDto nextHolder() {
+        if (!mIterator.hasNext()) {
+            if (proxyConfigHolders.isEmpty() || !infinityLooping) return null;
+            mIterator = proxyConfigHolders.iterator();
+        }
+        return mIterator.next();
+    }
+
```

Returning null is an anti-pattern. 
Wouldn't it be better to throw an exception immediately instead of checking if (holder == null) and throwing new RuntimeException("No more proxies"); in the getProxy() method?
It is also better to create a custom unchecked exception instead of it.